packages/experiments/src/components/Shimmer/Shimmer.base.tsx

```diff
@@ -0,0 +1,139 @@
+import { ShimmerCircle } from './ShimmerCircle/ShimmerCircle';
+
+const getClassNames = classNamesFunction<IShimmerStyleProps, IShimmerStyles>();
+
+export class ShimmerBase extends BaseComponent<IShimmerProps, {}> {
+  public static defaultProps: IShimmerProps = {
+    width: 100,
+    isDataLoaded: false
+  };
+  private _classNames: {[key in keyof IShimmerStyles]: string};
+  constructor(props: IShimmerProps) {
+    super(props);
+  }
+
+  public render(): JSX.Element {
+    const { getStyles, width, lineElements, children, isDataLoaded } = this.props;
+    const maxHeight: number | undefined = lineElements ? this._findMaxHeight(lineElements) : undefined;
+    this._classNames = getClassNames(getStyles!, { width, maxHeight, isDataLoaded });
+
+    const elements: JSX.Element[] | JSX.Element = lineElements ?
+      lineElements.map((elem: ICircle | ILine | IGap, index: number): JSX.Element => {
+        switch (elem.type) {
+          case ShimmerElementType.CIRCLE:
+            return (
+              <ShimmerCircle
+                key={ index }
+                { ...elem }
+                borderAlignStyle={ this._getBorderAlignStyles(maxHeight, elem) }
+              />
+            );
+          case ShimmerElementType.GAP:
+            const gapWidth = elem.width ? elem.width + '%' : '1%';
+            return (
+              <div
+                key={ index }
+                // tslint:disable-next-line:jsx-ban-props
+                style={ {
+                  width: gapWidth,
+                  height: maxHeight + 'px',
+                  backgroundColor: `${DefaultPalette.white}`
+                } }
+              />
+            );
+          case ShimmerElementType.RECTANGLE:
+            return (
+              <ShimmerRectangle
+                key={ index }
+                { ...elem }
+                borderAlignStyle={ this._getBorderAlignStyles(maxHeight, elem) }
+              />
+            );
+        }
+      }) :
+      <ShimmerRectangle />;
+
+    return (
+      <div className={ this._classNames.root }>
+        <div className={ this._classNames.shimmerWrapper }>
+          { elements }
+        </div>
+        <div className={ this._classNames.dataWrapper }>
+          { !!isDataLoaded ?
+            !!children ? children : null
+            : null }
+        </div>
+      </div>
+    );
+  }
+
+  private _findMaxHeight(items: Array<ICircle | IGap | ILine>): number {
+    const maxHeight = items.reduce((acc: number, next: ICircle | IGap | ILine): number => {
+      return next.height ?
+        next.height > acc ? next.height : acc
+        : acc;
+    }, 0);
+    return maxHeight;
+  }
+
+  private _getBorderAlignStyles(maxHeight: number | undefined, elem: ICircle | IGap | ILine): IStyleSet | undefined {
+    let height: number | undefined;
+    switch (elem.type) {
+      case ShimmerElementType.RECTANGLE:
+        height = !!elem.height ? elem.height : 16;
```

Comment:
I wonder if we should have these default values at a single source? Then the same value can used here and in defaultProps of the rectangle and circle components.